### ISSUE TITLE
lint suggestion: `needless_lifetimes` in `impl<'a>`

### ISSUE THREAD

**Issue by thetayloredman** [rust-lang/rust-clippy#12424]
### What it does

`needless_lifetimes` currently could support suggesting lifetime elison in `impl<'a> T<'a>` or `impl<'a> Trait for T<'a>` signatures.

### Advantage

Adds extra verbosity which is not necessary. If we elide lifetimes everywhere else, why don't we suggest it here?

### Drawbacks

Unsure if there is any.

### Example

```rust
impl<'a> MyType<'a> {
    // method definitions that don't reference 'a
}
```

Could be written as:

```rust
impl MyType<'_> {
    // method definitions that don't reference 'a
}
```

**Comment by m-rph** [comment]
Given that I am in the process of cleaning up `needless_lifetimes`, I will pick this up.

@rustbot claim

**Comment by y21** [comment]
This was implemented in #13286

**Comment by joshka** [comment]
I think this was missed from the changelog
